src/main/java/uk/co/codezen/maven/redlinerpm/rpm/RpmPackage.java

```diff
@@ -389,9 +414,14 @@ public void setFinalName(String finalName)
     public String getFinalName()
     {
         if (null == this.finalName) {
-            this.finalName = String.format("%s-%s-%s.%s.rpm",
-                    this.getName(), this.getVersion(), this.getRelease(),
-                    this.getArchitecture().toString().toLowerCase());
+            if (getOverrideName() == null) {
```

Comment:
and I guess also use this.overrideName instead of getOverrideName() here and below.